lib/service/storage/elasticsearch.js

```diff
@@ -2620,15 +2677,124 @@ class ElasticSearch extends Service {
   }
 
   /**
-   * Gets the name of an esIndex for an index + collection
+   * Given index + collection, returns the well formatted esIndex alias name
    *
    * @param {String} index
    * @param {String} collection
    *
-   * @returns {String} esIndex name (eg: '&nepali#liia')
+   * @returns {String} esIndex name (eg: '@&nepali.liia')
    */
   _getESIndex (index, collection) {
-    return `${this._indexPrefix}${index}${NAME_SEPARATOR}${collection}`;
+    return `${ALIAS_PREFIX}${this._indexPrefix}${index}${NAME_SEPARATOR}${collection}`;
+  }
+
+  /**
+   * Gets the concrete esIndex name from an alias (index + collection)
+   *
+   * @param {String} index
+   * @param {String} collection
+   *
+   * @returns {String} esIndex name (eg: '&nepali.liia')
+   * @throws If there is not exactly one esIndex related to this alias (index + collection)
+   */
+  async _getESIndexFromAlias (index, collection) {
+    const alias = `${ALIAS_PREFIX}${this._indexPrefix}${index}${NAME_SEPARATOR}${collection}`;
+    const { body } = await this._client.cat.aliases({ format: 'json', name: alias });
+
+    if (body.length < 1) {
+      throw kerror.get('unknown_index_collection');
+    }
+    else if (body.length > 1) {
+      throw kerror.get('multiple_index_alias', 'alias', 'index');
+    }
+
+    return body[0].index;
+  }
+
+  /**
+   * Gets the alias name from a concrete esIndex
+   *
+   * @param {String} esIndex
+   *
+   * @returns {String} alias name (eg: '@&nepali.liia')
+   * @throws If there is not exactly one alias related to this esIndex
+   */
+  async _getAliasFromESIndex (esIndex) {
+    const { body } = await this._client.indices.getAlias({ index: esIndex});
+    const aliases = Object.keys(body[esIndex].aliases);
+
+    if (aliases.length < 1) {
+      throw kerror.get('unknown_index_collection');
+    }
+    else if (aliases.length > 1) {
+      throw kerror.get('multiple_index_alias', 'index', 'alias');
+    }
+
+    return aliases[0];
+  }
+
+  /**
+   * Gets the concrete esIndex name from an alias (index + collection)
+   *
+   * @param {String} esIndex
+   *
+   * @returns {boolean} `true` if the raw ES index name is available, `false` otherwise
+   */
+  async _isESIndexAvailable (esIndex) {
+    return ! await this._client.indices.exists({ index: esIndex }).body;
+  }
+
+  /**
+   * Gets the concrete esIndex name from an alias (index + collection)
+   *
+   * @param {String} index
+   * @param {String} collection Default to `HIDDEN_COLLECTION`
+   *
+   * @returns {String} Available esIndex name (eg: '&nepali.liia')
+   */
+  async _getSafeESIndex (index, collection = HIDDEN_COLLECTION) {
```

Comment:
I don't get it why you put a default param here. IMHO it's more easier to understand if this method is always called with the same argument, a user collection or the hidden collection